analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -186,15 +186,30 @@ private static ProgramState ProcessAssertedBoolSymbol(ProgramState state, IOpera
         invocation switch
         {
             { Arguments.Length: 2, TargetMethod.IsStatic: true } => ProcessEquals(context, invocation.Arguments[0].ToArgument().Value, invocation.Arguments[1].ToArgument().Value),
-            { Arguments.Length: 1 } when invocation.TargetMethod.ContainingType.IsNullableValueType() => ProcessEquals(context, invocation.Instance, invocation.Arguments[0].ToArgument().Value),
+            { Arguments.Length: 1 } when invocation.TargetMethod.ContainingType.IsNullableValueType() || invocation.TargetMethod.ContainingType.IsStruct() =>
```

Comment:
I'm not super confident about this.
This is so we can process also invocations for `1.Equals(2)` and `false.Equals(true)`.
I limited it only to structs as for classes it might be overridden and create weird side efffects.

Reply:
Maybe I can be very specific here again and ask that the containing type is bool or int?